core/stablecog.py

```diff
@@ -15,7 +17,47 @@
 from core import viewhandler
 from core import settings
 from core import settingscog
+from threading import Thread
 
+async def update_progress(event_loop, status_message_task, s, queue_object, tries):
+    status_message = status_message_task.result()
+    try:
+        progress_data = s.get(url=f'{settings.global_var.url}/sdapi/v1/progress').json()
+
+        if progress_data["current_image"] is None and tries <= 10:
+            time.sleep(3)
+            event_loop.create_task(update_progress(event_loop, status_message_task, s, queue_object, tries + 1))
+            return
+
+        if progress_data["current_image"] is None and tries > 10:
+            return
+
+        image = Image.open(io.BytesIO(base64.b64decode(progress_data["current_image"])))
+
+        with contextlib.ExitStack() as stack:
+            buffer = stack.enter_context(io.BytesIO())
+            image.save(buffer, 'PNG')
+            buffer.seek(0)
+            file = discord.File(fp=buffer, filename=f'{queue_object.seed}.png')
+
+        ips = '?'
+        if progress_data["eta_relative"] != 0:
+            ips = round((int(queue_object.steps) - progress_data["state"]["sampling_step"]) / progress_data["eta_relative"], 2)
+
+        view = viewhandler.ProgressView()
+
+        await status_message.edit(
+            content=f'**Author**: {queue_object.ctx.author.id} ({queue_object.ctx.author.name})\n'
+                    f'**Prompt**: `{queue_object.prompt}`\n**Progress**: {round(progress_data.get("progress") * 100, 2)}% '
+                    f'\n{progress_data.get("state").get("sampling_step")}/{queue_object.steps} iterations, '
+                    f'~{ips} it/s'
+                    f'\n**ETA**: {round(progress_data.get("eta_relative"), 2)} seconds',
+            files=[file], view=view)
+    except Exception as e:
+        print('Something goes wrong...', str(e))
+
+    time.sleep(1)
+    event_loop.create_task(update_progress(event_loop, status_message_task, s, queue_object, tries))
```

Comment:
> I believe I am seeing this because my generation times are under ~8 seconds (using 3070 and SDXL Lightning) but the `update_progress` task waits at least 10 tries before finishing, regardless of whether the generation is done yet or not.

When there was a `time.sleep(1)` no errors but messages `404`
That's why [here ](https://github.com/Kilvoctu/aiyabot/pull/234#issuecomment-1965893650) I replaced 1 with 3
Bad choice.